docs/standard/base-types/parse-strings.md

```diff
@@ -0,0 +1,107 @@
+:::code language="visual-basic" source="snippets/parse-strings/vb/regex.vb" id="2":::
+
+The regular expression pattern `\[([^\[\]]+)\]` is defined like this:
+
+|Pattern|Description|
+|-------------|-----------------|
+|`\[`|Match an opening bracket.|
+|`([^\[\]]+)`|Match any character that is not an opening or a closing bracket one or more times. This is the first capturing group.|
+|`\]`|Match a closing bracket.|
+
+The <xref:System.Text.RegularExpressions.Regex.Split%2A?displayProperty=nameWithType> method is almost identical to <xref:System.String.Split%2A?displayProperty=nameWithType>, except that it splits a string based on a regular expression pattern instead of a fixed character set. For example, the following example uses the <xref:System.Text.RegularExpressions.Regex.Split%2A?displayProperty=nameWithType> method to split a string that contains substrings delimited by various combinations of hyphens and other characters.
+
+:::code language="csharp" source="snippets/parse-strings/csharp/regex.cs" id="3" interactive="try-dotnet":::
+:::code language="visual-basic" source="snippets/parse-strings/vb/regex.vb" id="3":::
+
+The regular expression pattern `\s-\s?[+*]?\s?-\s` is defined like this:
+
+|Pattern|Description|
+|-------------|-----------------|
+|`\s-`|Match a white-space character followed by a hyphen.|
+|`\s?`|Match zero or one white-space character.|
+|`[+*]?`|Match zero or one occurrence of either the + or * character.|
+|`\s?`|Match zero or one white-space character.|
+|`-\s`|Match a hyphen followed by a white-space character.|
+
+## String.IndexOf and String.Substring methods
+
+If you aren't interested in all of the substrings in a string, you might prefer to work with one of the string comparison methods that returns the index at which the match begins. You can then call the <xref:System.String.Substring%2A> method to extract the substring that you want. The string comparison methods include:
+
+- <xref:System.String.IndexOf%2A>, which returns the zero-based index of the first occurrence of a character or string in a string instance.
+
+- <xref:System.String.IndexOfAny%2A>, which returns the zero-based index in the current string instance of the first occurrence of any character in a character array.
+
+- <xref:System.String.LastIndexOf%2A>, which returns the zero-based index of the last occurrence of a character or string in a string instance.
+
+- <xref:System.String.LastIndexOfAny%2A>, which returns a zero-based index in the current string instance of the last occurrence of any character in a character array.
+
+The following example uses the <xref:System.String.IndexOf%2A> method to find the periods in a string. It then uses the <xref:System.String.Substring%2A> method to return full sentences.
+
+:::code language="csharp" source="snippets/parse-strings/csharp/indexof.cs" id="1" interactive="try-dotnet":::
+:::code language="visual-basic" source="snippets/parse-strings/vb/indexof.vb" id="1":::
```

Comment:
This ends abruptly; perhaps **see also** with link back to overview or some other string related examples?